src/components/Gallery.js

```diff
@@ -0,0 +1,40 @@
+import React, { Component } from "react";
+import { connect } from "react-redux";
+
+import { waitForImages } from "../utils/responsiveMasonry";
+
+class Gallery extends Component {
+    componentDidMount() {
+        waitForImages();
+    }
+    componentDidUpdate() {
+        waitForImages();
```

Comment:
Can you think of a way you would make this a bit more "React style"? It seems like the lazy loading is being deferred to a library called `imagesloaded`. I'm not familiar with this library, but it seems like it has caused some issues:
* It's not hooked into the react lifecycle - if you have a render between the time `waitForImages` begins and before it's finished, you have bugs
* You can't show a loading indicator in the image placeholder while the image is lazy loading



No? I might be wrong, just my first impression.